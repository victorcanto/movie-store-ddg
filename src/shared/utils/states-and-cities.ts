import statesAndCities from '../data/estados-cidades.json';

export const getBrazilianStates = () =>
	statesAndCities.estados.map(({ nome }) => nome);

export const getBrazilianCitiesByState = (stateName: string) =>
	statesAndCities.estados.find(({ nome }) => nome === stateName)?.cidades;
