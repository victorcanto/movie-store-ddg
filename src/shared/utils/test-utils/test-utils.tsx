import React, { PropsWithChildren } from 'react';
import { render as rtlRender } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import type { PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { setupStore, AppStore, RootState } from '../../../store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
	preloadedState?: PreloadedState<RootState>;
	store?: AppStore;
}

export function renderWithProviders(
	ui: React.ReactElement,
	{
		preloadedState = {},
		store = setupStore(preloadedState),
		...renderOptions
	}: ExtendedRenderOptions = {},
) {
	function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
		return (
			<Provider store={store}>
				<BrowserRouter>{children}</BrowserRouter>
			</Provider>
		);
	}
	return { store, ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';

export { renderWithProviders as render };
