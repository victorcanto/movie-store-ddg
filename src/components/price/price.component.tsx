import * as React from 'react';
import { Typography, TypographyProps } from '@mui/material';
import { currency } from '../../shared/utils/currency.utils';

export interface PriceProps extends TypographyProps {
	value: number;
}

export function Price({ value }: PriceProps) {
	return <span data-testid='price'>{currency.format(value)}</span>;
}
