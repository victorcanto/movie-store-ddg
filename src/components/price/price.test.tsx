import React from 'react';
import { render } from '../../shared/utils/test-utils';
import { Price } from './price.component';

test('snapshot of the base state', () => {
	const { asFragment } = render(<Price value={0} />);
	expect(asFragment()).toMatchSnapshot();
});

test('should have in the document', () => {
	const { getByTestId, getByText } = render(<Price value={0} />);
	expect(getByTestId('price')).toBeInTheDocument();
	expect(getByText('R$0.00')).toBeInTheDocument();
});
