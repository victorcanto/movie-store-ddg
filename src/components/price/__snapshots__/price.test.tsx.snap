// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of the base state 1`] = `
<DocumentFragment>
  <span
    data-testid="price"
  >
    R$0.00
  </span>
</DocumentFragment>
`;
